ar71xx: add support for D-Link DAP-1533

Tarask1n opened the pull request:
Access point D-Link DAP-1533:
- 4x 1Gbit port switch
- 1x 450Mbit 2.4/5GHz WiFi (3 internal antennas)
- 1x USB 2.0 port

It based on DB120 board:
- SoC Atheros AR7370 (AR9344 w/o WiFi) 
- 8 MiB flash
- 128 MiB ram 
- Ethernet switch AR8327N
- WiFi 2.4/5GHz AR9380

It could be easy converted to 1GBit router with Openwrt.
More detailed info I described there https://wiki.openwrt.org/toh/d-link/d-link_dap-1533_a1.

All works fine LED, buttons, Ethernet, WiFi, USB. But I've faced an issue. When I flashed an image to 16MiB flash the available space for software still left 3.7 Mb as an 8MiB flash. May be I've missed something in configuration.

Thanks in advance.

Signed-off-by: Taras Gavrylenko <gavrylenko.taras@gmail.com>

Tarask1n committed:
ar71xx: add support for D-Link DAP-1533

Add support for D-Link DAP-1533

Signed-off-by: Taras Gavrylenko <gavrylenko.taras@gmail.com>

heinzek reviewed (changes requested):
Hi @Tarask1n,

Please add commit message and add 'CONFIG_ATH79_MACH_DAP_1533_A1=y' to file 'target/linux/ar71xx/config-4.9'

Tarask1n commented:
Hi @heinzek,

I've done all changes that you requested, but compilation on project side was failed. On my side all Ok. Plaese see log:

```
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.000000] Linux version 4.9.73 (taras@taras-ubuntu) (gcc version 5.5.0 (OpenWrt GCC 5.5.0 r5683-3c7eec6) ) #0 Sun Jan 7 07:40:14 2018
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000] MyLoader: sysp=a140869b, boardp=59b43936, parts=b9d7c247
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] bootconsole [early0] enabled
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] CPU0 revision is: 0001974c (MIPS 74Kc)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] SoC: Atheros AR9344 rev 2
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Determined physical RAM map:
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000]  memory: 08000000 @ 00000000 (usable)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Initrd not found or empty - disabling initrd
Sun Jan  7 09:40:17 2018 kern.warn kernel: [    0.000000] OF: fdt:No valid device tree found, continuing without
Sun Jan  7 09:40:17 2018 kern.warn kernel: [    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
Sun Jan  7 09:40:17 2018 kern.warn kernel: [    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Zone ranges:
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Movable zone start for each node
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Early memory node ranges
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000] On node 0 totalpages: 32768
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000] free_area_init_node: node 0, pgdat 804827d4, node_mem_map 81000020
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000]   Normal zone: 256 pages used for memmap
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000]   Normal zone: 0 pages reserved
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000]   Normal zone: 32768 pages, LIFO batch:7
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.000000] pcpu-alloc: [0] 0
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 32512
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.000000] Kernel command line:  board=DAP-1533-A1 mtdparts=spi0.0:128k(uboot)ro,64k(nvram)ro,7680k(firmware),192k(lang)ro,64k(mac)ro,64k(art)ro console=ttyS0,115200 rootfstype=squashfs noinitrd
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Writing ErrCtl register=00000000
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Readback ErrCtl register=00000000
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Memory: 124744K/131072K available (3245K kernel code, 167K rwdata, 824K rodata, 300K init, 213K bss, 6328K reserved, 0K cma-reserved)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] NR_IRQS:51
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] Clocks: CPU:550.000MHz, DDR:400.000MHz, AHB:200.000MHz, Ref:40.000MHz
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6950037990 ns
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.000010] sched_clock: 32 bits at 275MHz, resolution 3ns, wraps every 7809031678ns
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.008710] Calibrating delay loop... 274.02 BogoMIPS (lpj=1370112)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.081674] pid_max: default: 32768 minimum: 301
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.086934] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.094321] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.104842] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.115841] futex hash table entries: 256 (order: -1, 3072 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.123235] NET: Registered protocol family 16
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.129773] MIPS: machine is D-Link DAP-1533 rev. A1
Sun Jan  7 09:40:17 2018 kern.warn kernel: [    0.137319] registering PCI controller with io_map_base unset
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.401971] PCI host bridge to bus 0000:00
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.406604] pci_bus 0000:00: root bus resource [mem 0x10000000-0x13ffffff]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.414270] pci_bus 0000:00: root bus resource [io  0x0000]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.420494] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.428058] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.436932] pci 0000:00:00.0: [168c:0030] type 00 class 0x028000
Sun Jan  7 09:40:17 2018 kern.err kernel: [    0.436943] pci 0000:00:00.0: invalid calibration data
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.442692] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x0001ffff 64bit]
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.442745] pci 0000:00:00.0: reg 0x30: [mem 0x00000000-0x0000ffff pref]
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.442823] pci 0000:00:00.0: supports D1
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.442833] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.443060] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.443089] pci 0000:00:00.0: BAR 0: assigned [mem 0x10000000-0x1001ffff 64bit]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.451290] pci 0000:00:00.0: BAR 6: assigned [mem 0x10020000-0x1002ffff pref]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.459352] pci 0000:00:00.0: using irq 40 for pin 1
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.465632] clocksource: Switched to clocksource MIPS
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.472382] NET: Registered protocol family 2
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.478125] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.485963] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.493059] TCP: Hash tables configured (established 1024 bind 1024)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.500250] UDP hash table entries: 256 (order: 0, 4096 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.506811] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.514066] NET: Registered protocol family 1
Sun Jan  7 09:40:17 2018 kern.debug kernel: [    0.519021] PCI: CLS 0 bytes, default 32
Sun Jan  7 09:40:17 2018 kern.warn kernel: [    0.523327] Crashlog allocated RAM at address 0x3f00000
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.530190] workingset: timestamp_bits=30 max_order=15 bucket_order=0
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.544717] squashfs: version 4.0 (2009/01/31) Phillip Lougher
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.551262] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.570600] io scheduler noop registered
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.574977] io scheduler deadline registered (default)
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.581206] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.588787] console [ttyS0] disabled
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.612861] serial8250.0: ttyS0 at MMIO 0x18020000 (irq = 11, base_baud = 2500000) is a 16550A
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.622495] console [ttyS0] enabled
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.629915] bootconsole [early0] disabled
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.642262] m25p80 spi0.0: mx25l6405d (8192 Kbytes)
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.647315] 6 cmdlinepart partitions found on MTD device spi0.0
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.653322] Creating 6 MTD partitions on "spi0.0":
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.658209] 0x000000000000-0x000000020000 : "uboot"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.665250] 0x000000020000-0x000000030000 : "nvram"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.671951] 0x000000030000-0x0000007b0000 : "firmware"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.688772] 2 uimage-fw partitions found on MTD device firmware
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.694800] 0x000000030000-0x000000190000 : "kernel"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.701292] 0x000000190000-0x0000007b0000 : "rootfs"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.708048] mtd: device 4 (rootfs) set to be root filesystem
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.713829] 1 squashfs-split partitions found on MTD device rootfs
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.720162] 0x0000003d0000-0x0000007b0000 : "rootfs_data"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.727365] 0x0000007b0000-0x0000007e0000 : "lang"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.733907] 0x0000007e0000-0x0000007f0000 : "mac"
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    0.740435] 0x0000007f0000-0x000000800000 : "art"
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.748018] libphy: Fixed MDIO Bus: probed
Sun Jan  7 09:40:17 2018 kern.info kernel: [    0.783663] libphy: ag71xx_mdio: probed
Sun Jan  7 09:40:17 2018 kern.info kernel: [    1.417167] ag71xx ag71xx.0: connected to PHY at ag71xx-mdio.0:00 [uid=00000000, driver=Generic PHY]
Sun Jan  7 09:40:17 2018 kern.info kernel: [    1.427189] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode:RGMII
Sun Jan  7 09:40:17 2018 kern.info kernel: [    1.435122] NET: Registered protocol family 10
Sun Jan  7 09:40:17 2018 kern.info kernel: [    1.442996] NET: Registered protocol family 17
Sun Jan  7 09:40:17 2018 kern.info kernel: [    1.447737] 8021q: 802.1Q VLAN Support v1.8
Sun Jan  7 09:40:17 2018 kern.info kernel: [    1.457647] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
Sun Jan  7 09:40:17 2018 kern.info kernel: [    1.466475] Freeing unused kernel memory: 300K
Sun Jan  7 09:40:17 2018 kern.warn kernel: [    1.470992] This architecture does not have kernel memory protection.
Sun Jan  7 09:40:17 2018 user.info kernel: [    2.209918] init: Console is alive
Sun Jan  7 09:40:17 2018 user.info kernel: [    2.213605] init: - watchdog -
Sun Jan  7 09:40:17 2018 kern.notice kernel: [    2.525646] random: fast init done
Sun Jan  7 09:40:17 2018 user.info kernel: [    3.213642] kmodloader: loading kernel modules from /etc/modules-boot.d/*
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.284525] usbcore: registered new interface driver usbfs
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.290252] usbcore: registered new interface driver hub
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.295779] usbcore: registered new device driver usb
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.306613] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.314974] ehci-platform: EHCI generic platform driver
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.320440] ehci-platform ehci-platform: EHCI Host Controller
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.326337] ehci-platform ehci-platform: new USB bus registered, assigned bus number 1
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.336474] ehci-platform ehci-platform: irq 3, io mem 0x1b000000
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.365674] ehci-platform ehci-platform: USB 2.0 started, EHCI 1.00
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.372945] hub 1-0:1.0: USB hub found
Sun Jan  7 09:40:17 2018 kern.info kernel: [    3.377192] hub 1-0:1.0: 1 port detected
Sun Jan  7 09:40:17 2018 user.info kernel: [    3.382040] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
Sun Jan  7 09:40:17 2018 user.info kernel: [    3.399867] init: - preinit -
Sun Jan  7 09:40:17 2018 kern.info kernel: [    4.049900] usb 1-1: new high-speed USB device number 2 using ehci-platform
```

pepe2k commented on target/linux/ar71xx/files/arch/mips/ath79/mach-dap-1533-a1.c:
```diff
@@ -0,0 +1,180 @@
+
+static struct gpio_led dap1533a1_leds_gpio[] __initdata = {
+	{
+		.name		= "d-link:green:power",
```
Please, use "board:color:function" LED names (don't forget to update them in user space scripts).

pepe2k commented on target/linux/ar71xx/files/arch/mips/ath79/mach-dap-1533-a1.c:
```diff
@@ -0,0 +1,180 @@
+		.active_low		= 1,
+	},
+	{
+		.desc		= "WPS button",
```
Please, align indentation (you have different one above).

pepe2k commented on target/linux/ar71xx/files/arch/mips/ath79/mach-dap-1533-a1.c:
```diff
@@ -0,0 +1,180 @@
+	},
+};
+
+static struct flash_platform_data dap1533a1_flash_data = {
```
This shouldn't be required, please drop it.

pepe2k commented on target/linux/ar71xx/image/generic.mk:
```diff
@@ -1083,3 +1083,14 @@ define Device/fritz300e
+  DEVICE_TITLE := D-Link DAP-1533 rev. A1
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2
+  BOARDNAME := DAP-1533-A1
+  SUPPORTED_DEVICES := dap-1533-a1
```
Your image doesn't use metadata, drop this line.

pepe2k commented on target/linux/ar71xx/image/generic.mk:
```diff
@@ -1083,3 +1083,14 @@ define Device/fritz300e
+  BOARDNAME := DAP-1533-A1
+  SUPPORTED_DEVICES := dap-1533-a1
+  IMAGE_SIZE := 7680k
+  IMAGE/sysupgrade.bin := append-kernel | pad-to 64k | append-rootfs | pad-rootfs | check-size $$$$(IMAGE_SIZE)
```
This is default recipe, drop this line please.

pepe2k commented on target/linux/ar71xx/files/arch/mips/ath79/mach-dap-1533-a1.c:
```diff
@@ -0,0 +1,180 @@
+	ath79_eth0_data.phy_if_mode = PHY_INTERFACE_MODE_RGMII;
+	ath79_eth0_data.phy_mask = BIT(0);
+	ath79_eth0_data.mii_bus_dev = &ath79_mdio0_device.dev;
+	ath79_eth0_pll_data.pll_1000 = 0x06000000;
```
Are you sure about this value? Have you made some throughput tests to check if RGMII works correctly? Can you check what value sets vendor firmware/U-Boot?

pepe2k reviewed (changes requested):
Hello @Tarask1n,

Your commit description (not Pull Request message) should contain hardware description and flashing instruction.

Also, please have a look at my comments inline, below.

Cheers,
Piotr

mkresin commented:
Please reopen the PR if you had the time to address the requested changes.
